### DISCUSSION TITLE
Is there an idle inhibitor widget?

### DISCUSSION THREAD

**Discussion by kode54** [nwg-piotr/nwg-shell#112] [Q&A]
As asked, is there a widget to do this, similar to what Waybar already provides?

**Comment by nwg-piotr** [comment]
I never needed any. Is it a feature request?

**Comment by kode54** [comment]
I suppose it is. I needed it when playing a game with a game controller, during which swayidle first attempted to lock my screen (which failed for some reason, I'll have to determine why later), and then the wlopm --off \* kicked in, which powered off my displays.

I suppose it would also be possible for an idle monitor to check on game controller type libinput devices. It would probably be a first for a desktop environment.

**Reply by nwg-piotr** [reply]
man 5 sway

`inhibit_idle focus|fullscreen|open|none|visible`

Set/unset an idle inhibitor for the view. focus will inhibit idle when the view is focused by any seat. fullscreen will inhibit idle when the view is fullscreen (or a descendant of a fullscreen container) and is visible. open will inhibit idle until the view is closed (or the inhibitor is unset/changed). visible will inhibit idle when the view is visible on any output. none will remove any existing idle inhibitor for the view.
This can also be used with criteria to set an idle inhibitor for any existing view or with for_window to set idle inhibitors for future views.

**Reply by kode54** [reply]
And I am not using sway, but labwc. I shall continue using waybar instead for now.
